.github/workflows/build-plugin-archive.yml

```diff
@@ -118,38 +104,82 @@ jobs:
+      - name: Install Composer dependencies without dev dependencies
         uses: ramsey/composer-install@v3
         with:
-          composer-options: --prefer-dist
+          composer-options: ${{ inputs.COMPOSER_ARGS }}
+
+      - name: Set artifact name
+        id: set-artifact-name
+        run: echo "artifact=interim-deps" >> $GITHUB_OUTPUT
 
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ steps.set-artifact-name.outputs.artifact }}
+          path: |
+            ./*
+            !./.git
+            !./.ddev
+            !./.github
+  run-build-tools:
+    name: Process build steps
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    needs: checkout-dependencies
+    env:
+      NODE_OPTIONS: ${{ inputs.NODE_OPTIONS }}
+      NODE_AUTH_TOKEN: ${{ secrets.NPM_REGISTRY_TOKEN }}
+      NODE_CACHE_MODE: ''
+    outputs:
+      artifact: ${{ steps.set-artifact-name.outputs.artifact }}
+    steps:
+      - name: Download Artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ needs.checkout-dependencies.outputs.artifact }}
       - name: Check optional Composer build tools
         id: composer-tools
         run: |
+          hasAssetConfig(){
+            test -f assets-compiler.json
+            local EXIT=$?
+            if [ ! $EXIT ]; then
+              echo "$EXIT"
+              exit 0
+            fi
+            jq '.extra | has("composer-asset-compiler")' < composer.json >/dev/null 2>&1
+            local EXIT=$?
+            echo "$EXIT"
+          }
+          hasTranslateConfig(){
+            jq '.extra | has("wp-translation-downloader")' < composer.json >/dev/null 2>&1
+            local EXIT=$?
+            echo "$EXIT"
+          }
           hasDep(){
             composer show -i -N -D --strict "${1}" >/dev/null 2>&1
             local EXIT=$?
             echo "$EXIT"
           }
-          echo "assets-compiler=$( hasDep inpsyde/composer-assets-compiler )" >> $GITHUB_OUTPUT
-          echo "translation-downloader=$( hasDep inpsyde/wp-translation-downloader )" >> $GITHUB_OUTPUT
+          echo "assets-compiler=$( hasAssetConfig )" >> $GITHUB_OUTPUT
+          echo "translation-downloader=$( hasTranslateConfig )" >> $GITHUB_OUTPUT
+
+      - name: Set up node cache mode
+        run: |
+          if [ "${{ inputs.PACKAGE_MANAGER }}" == 'npm' ] && { [ -f "${GITHUB_WORKSPACE}/package-lock.json" ] || [ -f "${GITHUB_WORKSPACE}/npm-shrinkwrap.json" ]; }; then
+            echo "NODE_CACHE_MODE=npm" >> $GITHUB_ENV
+          elif [ "${{ inputs.PACKAGE_MANAGER }}" == 'yarn' ] && [ -f "${GITHUB_WORKSPACE}/yarn.lock" ]; then
+            echo "NODE_CACHE_MODE=yarn" >> $GITHUB_ENV
+          else
+            echo "No lock files found or unknown package manager"
+          fi
 
       - name: Check for package.json
```

Comment:
I know this is existing code, but why do we need this step? We already check for the presence of the Composer Asset Compiler configuration; there's no valid reason for it to be present in a package that does _not_ have a `package.json` file.

Also, the `check-for-package` step ID just confused me while reading the code. It is always a package; what we do here is to check if it is an _npm_ package or rather if it has a `package.json` file. If you decide to keep this step, can we at least improve the ID?

Reply:
I removed it. Thanks for pointing this out